library/src/main/java/com/novoda/downloadmanager/lib/OkHttpNotificationImageRetriever.java

```diff
@@ -36,4 +49,14 @@ public Bitmap retrieveImage(String imageUrl) {
             return null;
         }
     }
+
+    private void cleanupOldBitmap() {
+        if (bitmap == null) {
+            return;
+        }
+        if (!bitmap.isRecycled()) {
+            bitmap.recycle();
```

Comment:
plus the problem is you return that bitmap in `retrieveImage` which means someone else could be using that bitmap so if you recycle it here and that bitmap is still in use an exception will be thrown